src/FD/grid/surfaces_interface.f90

```diff
@@ -50,6 +50,13 @@ module subroutine set_halo_data(my_halo_data)
       class(package), intent(in), dimension(:,:,:) :: my_halo_data
     end subroutine
 
+    pure module function get_halo_data() result(singleton_halo_data)
+      !! result contains halo-exchange data packaged in an object array with dimenions
+      !! [my_blocks(first):my_blocks(last), space_dimension, size([forward, backward]))
```

Comment:
@rouson I think there's a missing closing `]` here. Also, code snippets should be in backticks to properly render documentation
```suggestion
      `!! [my_blocks(first):my_blocks(last), space_dimension, size([forward, backward]))]`
```

Reply:
I'm going to commit this and merge.